teps/0044-decouple-task-composition-from-scheduling.md

```diff
@@ -84,97 +84,399 @@ that can be shared between them, and running the second Task will be delayed by
+        name: just-unit-tests
+      workspaces:
+        - name: source-code
+          workspcae: source-code
+        - name: test-results
+          workspace: test-results
+  finally:
+    - name: upload-results
+      runAfter: run-unit-tests
+      taskRef:
+        name: gcs-upload
+      params:
+        - name: location
+          value: gs://my-test-results-bucket/testrun-$(taskRun.name)
+      workspaces:
+        - name: data
+          workspace: test-results
+    - name: update-slack
+      params:
+        - name: message
+          value: "Tests completed with $(tasks.run-unit-tests.status) status"
+```
+
+We could leverage
+[TEP-0056 pipelines in pipelines](https://github.com/tektoncd/community/blob/main/teps/0056-pipelines-in-pipelines.md):
+to run this Pipeline within another as a pod, for example:
+
+
+```yaml
+kind: Pipeline
+metadata:
+  name: build-test-deploy
+spec:
+  workspaces:
+    - name: source-code
+  tasks:
+    - name: get-source
+      workspaces:
+        - name: source-code
+          workspace: source-code
+      taskRef:
+        name: git-clone
+      params:
+        - name: url
+            value: $(params.url)
+        - name: revision
+          value: $(params.revision)
+    - name: run-unit-tests-and-upload-results
+      runAfter: get-source
+      pipelineRef: # syntax may look different as TEP-0056 but the idea is that this is a Pipeline (in a pipeline) instead of a Task or Run
+        name: unit-tests
+      execution/scheduling: "co-located" # this pipeline in a pipeline should run in a pod
+      workspaces:
+        - name: source-code
+          workspcae: source-code
+    - name: deploy
+      runAfter: [run-unit-tests-and-upload-results]
+      when:
+        - input: "$(params.git-branch)"
+          operator: in
+          values: ["main"]
+      taskRef:
+        name: build-deploy
+      workspaces:
+        - name: source-code
+          workspcae: source-code
+```
+
+In the above Pipeline, the Task `git-clone` would run as a pod, the Pipeline `unit-test` would be executed as a pod
+(by creating a PipelineRun using the new syntax) - including two `finally` Tasks, and the Task `build-deploy` would run
+as a pod.
 
+The Pipeline `unit-test` also takes a workspace `test-results` but when executing as a pod, this can be satisfied by
+`emptyDir`, so in the Pipeline above that runs `unit-test` as a Pipeline in a pod, it skips specifying this and since
+the execution mode is `co-located` the PipelineRun can be constructed to pass `emptyDir` (or that could be the
+implied default for any workspaces when a PipelineRun is run in `co-located` mode).
 
-### Task composition in Pipeline Tasks
+Pros:
+* Doesn't involve introduction of new types
+* Assuming
+  [TEP-0056 pipelines in pipelines](https://github.com/tektoncd/community/blob/main/teps/0056-pipelines-in-pipelines.md)
+  users can construct composition of Pipelines however they'd like (i.e. parts of a Pipeline in a pod, parts not)
```

Comment:
Isn't it independent though ? If you have an option to run Pipeline in a Pod through a PipelineRun option, having pipelines in pipelines built-in Pipeline or not, we would be able to use that feature wouldn't we ?